### ISSUE TITLE
Create User Preference/Settings Page 

### ISSUE THREAD

**Issue by ritwikshanker** [ritwikshanker/GoCoronaGo#65]
As of now, only changing theme's is the customization available to the user. 
We can also add his preferred country so that they don't have to always select their country from the dropdown. 
For that, we will need a separate screen so in future, if we want we can increase more customization options. 
In layman terms, we sort of have to build a **settings** page.

**Comment by ritwikshanker** [comment]
Also, we already have strings in 4 languages so we can add app language option for wider reach.

**Comment by hardiksachan** [comment]
Hey! I would like to work on this, can you please assign this to me. Also, I'm thinking of using Jetpack Datastore to save user preferences, is that okay?

**Comment by ritwikshanker** [comment]
> Hey! I would like to work on this, can you please assign this to me. Also, I'm thinking of using Jetpack Datastore to save user preferences, is that okay?

Sure @hardiksachan, Feel free to use the datastore. 
- Currently, I'm working on the API that will be providing country-related data, so, for now, add an option in the data store to save the preferred country, I'll pick that up on my home screen and compute accordingly. 
- Also, please follow MVVM architecture throughout the screen. 
- Also, I'm already working on changing the language code, so you can just implement the code to add multiple languages in a `dropdown/Dialogfragment` and save it in the data store according to user choice and I'll merge my code that will change the locale of the app. 
- Also you might have to consider the case of user data migration from SharedPreferences to Datastore for current users.

**Comment by hardiksachan** [comment]
Okay @ritwikshanker, This is what I'll be working on.

* I'll create a repository to save the following in data store
         1. Preferred Country
         2. Preferred Language
         3. Preferred App Theme
* I will follow MVVM, though I would like if we don't use interactor/usecases, I think it's too much abstraction, is that okay?
* RE: Migration from SharedPreferences to Datastore, I'll create a separate issue for it, and work on it later.
* I will write unit tests, but not integration tests.

Also, I have a question - I'll be creating enums to represent available countries and languages, right?

**Comment by ritwikshanker** [comment]
@hardiksachan 
- First point is fine
- Yes don't use interactors/use cases
- This is also fine. Create an issue separately
- If you want to write test cases it'll be great but those are not mandatory, so without any test case, your PR will be GTG.
- For languages yes, for countries, API will supply data and I will populate the list accordingly. You just need to store the preferred country in the data store. For list we will be using this API - https://disease.sh/docs/#/

**Comment by hardiksachan** [comment]
Alright, I'll save countries as a string. I'll raise a PR as soon as I'm done.

**Comment by hardiksachan** [comment]
I'm using these libraries for testing at the moment:
- JUnit4
- Truth
- Mockk

**Comment by hardiksachan** [comment]
Hey @ritwikshanker, I have implemented datastore and a settings page from where the language, country, and theme can be updated.

But, I have not wired up the logic to actually observe the settings and update the app theme because currently there are two sources of truth for the theme - XML views, and compose theme.

Since, the complete app is being ported to Compose, can we do it after the migration is done? And should I raise a PR?
